### PR TITLE
added echart binding

### PR DESCRIPTION
**Pull request by bruckwubete**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [x] #major# - incompatible API change

# CHANGELOG
* **Added** echarts binding

### CODE REVIEW

**Comment by bruckwubete** [comment]
link to echarts: https://ecomfe.github.io/echarts-examples/public/index.html

##Results in the dummy app

screen shot 2018-03-05 at 2 34 01 pm


screen shot 2018-03-05 at 2 33 34 pm

File: package.json [hunk 1]
```diff
@@ -50,6 +50,9 @@
     "frost"
   ],
   "dependencies": {
+    "broccoli-funnel": "^2.0.1",
+    "broccoli-merge-trees": "^3.0.0",
```
Comment (juwara0) [inline package.json]: This needs to be `^2.0.0`

File: vendor/shims/echarts.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+(function() {
+  function vendorModule() {
+    'use strict';
```
Comment (juwara0) [inline vendor/shims/echarts.js]: Please remove the semicolons in this file.

File: vendor/shims/echarts.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+    return { 'default': self['echarts'] };
```
Comment (juwara0) [inline vendor/shims/echarts.js]: Please remove the semicolons in this file.

File: vendor/shims/echarts.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  }
+
+  define('echarts', [], vendorModule);
```
Comment (juwara0) [inline vendor/shims/echarts.js]: Please remove the semicolons in this file.

File: vendor/shims/echarts.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+})();
```
Comment (juwara0) [inline vendor/shims/echarts.js]: Please remove the semicolons in this file.

**Comment by gkchestertron** [comment]
@bruckwubete can you link this to an issue or at least describe why you are adding echarts to this addon? This seems like a lot to add to this addon's footprint if there is not a need for it across the organization. Especially considering the dependency issues it will likely cause.

**Review by notmessenger — CHANGES_REQUESTED** [review]
@bruckwubete Can you please provide background information on why these changes are being proposed? What problem is being solved, what ticket being addressed, etc?  What other solutions were considered?

It will likely be more productive for us to schedule a video chat to discuss this in depth.

File: index.js [hunk 6]
```diff
@@ -15,6 +19,20 @@ module.exports = {
...
   /* eslint-enable complexity */
+
+  included () {
+    this._super.included.apply(this, arguments)
+    this.import('vendor/echarts-en.min.js')
+    this.import('vendor/shims/echarts.js')
+  },
+
+  treeForVendor (vendorTree) {
+    const momentTree = new Funnel(path.join(this.project.root, 'node_modules', 'echarts', 'dist'), {
```
Comment (juwara0) [inline index.js]: momentTree?

File: index.js [hunk 7]
```diff
@@ -15,6 +19,20 @@ module.exports = {
...
+      files: ['echarts-en.min.js']
+    })
+
+    return new MergeTrees([vendorTree, momentTree])
```
Comment (juwara0) [inline index.js]: momentTree?

File: index.js (hunk 6 above)
Comment (gkchestertron) [inline index.js]: moment?

**Review by gkchestertron — CHANGES_REQUESTED** [review]

**Comment by bruckwubete** [comment]
Adding this change for potential use of majority of these echarts for vectors 2018 demo. We're probably gonna build a logging and metrics app just like kibana. So we need a way to easily render some pretty charts.  @gkchestertron @notmessenger 

And @gkchestertron echarts is written in pure JavaScript. It has only two dependencies  "chartjs-color": "^2.1.0" and "moment": "^2.10.2"

File: index.js (hunk 7 above)
Comment (bruckwubete) [reply index.js]: I'll rename to echartTree. Was following a moment guide of exposing npm libraries as a shim

File: index.js (hunk 6 above)
Comment (bruckwubete) [reply index.js]: I'll rename to echartTree. Was following a moment guide of exposing npm libraries as a shim

Comment (bruckwubete) [reply index.js]: I'll rename to echartTree. Was following a moment guide of exposing npm libraries as a shim

**Comment by notmessenger** [comment]
@bruckwubete The shimming of https://github.com/ecomfe/echarts should not be done as part of this repository but rather in its own.  Give me a little bit of time this morning and I will get a new repo setup in which these changes can be applied and will provide you the details.

**Comment by notmessenger** [comment]
@bruckwubete Actually https://www.npmjs.com/package/ember-echarts already exists in the ecosystem so there is no need for us to create our own shim for the echarts library.   Please use https://www.npmjs.com/package/ember-echarts for your projects needs.

**Comment by bruckwubete** [comment]
that repo was ill maintained so I did it myself :)  https://github.com/bruckwubete/echarts-ember

**Pull request closed without merging**